### PR TITLE
Grammar: Reduce and rename for clarity

### PR DESCRIPTION
**Pull request by mcsf**
## Description

Spin-off from #6116.

Grammar changes:
* Rename `Token` to `Block`
* Rename `Block_Void` to `Block_Empty`
* Port `WS+` to `__`, merging together white-space rules

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style.
- [ ] My code has proper inline documentation.

### CODE REVIEW

File: blocks/api/post.pegjs [hunk 1]
```diff
@@ -161,19 +161,19 @@ function partition( predicate, list ) {
...
-  { /** <?php return peg_join_blocks( $pre, $ts, $post ); ?> **/
-    return joinBlocks( pre, ts, post );
+  { /** <?php return peg_join_blocks( $pre, $bs, $post ); ?> **/
+    return joinBlocks( pre, bs, post );
   }
 
-Token
-  = Block_Void
+Block
+  = Block_Empty
```
Comment (dmsnell) [inline blocks/api/post.pegjs]: It’s worth considering the context in which void was chosen as a name. It borrows from the HTML spec which specifies certain tags as void and those are not allowed to have a closing tag - hr, img, br, etc... in Gutenberg it’s optional the distinction between empty and void but the intention was to convey void elements, not to present an optimization for empty elements

**Review by dmsnell — APPROVED** [review]
I should clarify what I was saying about `Void`. I approve these changes though I think it may still be worth considering _not_ changing `Void` to `Empty` because they carry different semantic nuances.

Make your choice; I support this either way.

**Comment by mcsf** [comment]
Thanks for the review and confidence, Dennis. Your comment on void/empty makes a lot of sense.

I'll keep `Block_Void`, then. What I think we'll do with/after #6116 is add comments to the rules, and one for `Block_Void` would include your HTML-spec-inspired reasoning, and that it "represents an empty block", so to speak. Whatever makes the grammar doc easier to read.

**Comment by dmsnell** [comment]
The point is that void blocks aren’t _empty_ blocks but thy are blocks that cannot contain content. Maybe we muddy that in our parser/serializer. If it ends up that actually we use it as a shorthand then we may actually want to rename it. Nice work in any case!

**Pull request merged**